ext/UnitfulExt.jl

```diff
@@ -200,27 +197,14 @@ Base.ncodeunits(n::S) = ncodeunits(n.content)
 Base.isvalid(n::S, i::Integer) = isvalid(n.content, i)
 Base.pointer(n::S) = pointer(n.content)
 Base.pointer(n::S, i::Integer) = pointer(n.content, i)
-"""
-    P_str(s)
 
-Creates a string that will be Protected from recipe passes.
-
-Example:
-```julia
-julia> plot([0,1]u"m", [1,2]u"m/s^2", xlabel=P"This label will NOT display units")
-
-julia> plot([0,1]u"m", [1,2]u"m/s^2", xlabel="This label will display units")
-```
-"""
-macro P_str(s)
-    return ProtectedString(s)
-end
+Plots.protectedstring(s) = ProtectedString(s)
```

Comment:
why does the macro needs to be moved?

Reply:
It doesn't seem possible to export symbols from the weak extensions modules.

So the "fix" here I found is to define a `protectedstring` helper in Plots (generic function with no method) and define a method `Plots.protectedstring` in the weak ext module.

Reply:
would it work if we only have `macro P_str end` in Plots and then the implementation in the extension?

Probably macros don't hade a methods table, right?

Reply:
You cannot do something like this:
```julia
macro Plots.P_str(...)
    # ...
end
```
And using the following doesn't work either (breaks `UnitfulExt` precompilation):
```julia
@eval Plots macro P_str(...)
    # ...
end
```

Reply:
> Finally, this doesn't work (no `P_str` symbol exported):
> 
> ```julia
> export P_str
> macro P_str(...)
>     # ...
> end
> ```

I would expect that this exports to `Plots` rather than Main